### DISCUSSION TITLE
How do I logout all the logged in Users.

### DISCUSSION THREAD

**Discussion by MikePooh** [cutelyst/cutelyst#322] [Q&A] [answered]
Hi!

I am using Session and Authentication plugins of Cutelyst framework . In case of some event I need to logout **all** the users and force them to login page. Whats the right way to do so?

**Comment by dantti** [comment]
You would need to clear all session, if you are using memcached for storing session info you can do a server restart, if it's files them delete them.

**Reply by MikePooh** [reply]
I've tested the following:

1. Made the function in /Cutelyst/Plugins/Session/sessionstorefile.h
```
static QString rootPath()
{
    return QDir::tempPath()
            + QLatin1Char('/')
            + QCoreApplication::applicationName()
            + QLatin1String("/session/data");
}
```
2. Than in my application I am able to call
```
if(dynamic_cast<SessionStoreFile*>(app->plugin<Session*>()->storage()))
    {
        QDir dir(SessionStoreFile::rootPath());
        dir.removeRecursively();
    } 
```
If you accept this approach I can make a PR with this minor changes.

**Reply by dantti** [reply]
I think the right approach would be to that implement in the stored called by Session::deleteExpiredSessions(), so that a 0 value would clean all.

**Reply by MikePooh** [reply]
I see. Will make the PR soon.

**Comment by dantti** [comment]
Another option is to store a session version so that you increase the number and logout the user if using the old version.

**Comment by MikePooh** [comment, marked as answer]
After merging this [PR](https://github.com/cutelyst/cutelyst/commit/5561c43faceefd77ba63811c0de30e254545b961) now its possible to do:
`app->plugin<Session*>()->storage()->deleteExpiredSessions(nullptr, 0);`
Zero as last argument will remove all sessions.
This is suitable for SessionStoreFile only.
